File: application/src/test/java/com/study/platform/support/fake/FakeApplyQueryPort.java
package com.study.platform.support.fake;

import com.study.platform.domain.apply.dto.response.ApplyResponse;
import com.study.platform.domain.apply.port.ApplyQueryPort;

import java.util.List;
import java.util.UUID;

public class FakeApplyQueryPort implements ApplyQueryPort {

    private final FakeApplyRepository repository;

    public FakeApplyQueryPort(FakeApplyRepository repository) {
        this.repository = repository;
    }

    @Override
    public List<ApplyResponse> findAllByPostId(UUID postId) {
        return repository.findAllByPostId(postId).stream()
                .map(ApplyResponse::from)
                .toList();
    }
}

File: application/src/test/kotlin/com/study/platform/support/fake/FakeApplyQueryPort.kt
package com.study.platform.support.fake

import com.study.platform.domain.apply.dto.response.ApplyResponse
import com.study.platform.domain.apply.port.ApplyQueryPort
import org.springframework.data.domain.Page
import org.springframework.data.domain.PageImpl
import org.springframework.data.domain.Pageable
import java.util.UUID

class FakeApplyQueryPort(
    private val repository: FakeApplyRepository
) : ApplyQueryPort {

    override fun findAllByPostId(postId: UUID, pageable: Pageable): Page<ApplyResponse> {
        val all = repository.findAllByPostId(postId)
            .sortedByDescending { it.createdAt }
            .map { ApplyResponse.from(it) }
        val content = all.drop(pageable.offset.toInt()).take(pageable.pageSize)
        return PageImpl(content, pageable, all.size.toLong())
    }
}
